mlflow/store/tracking/sqlalchemy_store.py

```diff
@@ -1913,6 +1912,128 @@ def _apply_order_by_search_logged_models(
 
         return models.order_by(*order_by_clauses)
 
+    def _apply_filter_string_search_logged_models(
+        self,
+        models: sqlalchemy.orm.Query,
+        session: sqlalchemy.orm.Session,
+        experiment_ids: list[str],
+        filter_string: Optional[str],
+    ):
+        import sqlparse
+
+        if not filter_string:
+            return session.query(SqlLoggedModel).filter(
+                SqlLoggedModel.experiment_id.in_(experiment_ids)
+            )
+
+        try:
+            parsed = sqlparse.parse(filter_string)
+        except Exception as e:
+            raise MlflowException.invalid_parameter_value(
+                f"Invalid filter string: {filter_string}. {e!r}"
+            ) from e
+
+        if len(parsed) != 1:
+            raise MlflowException.invalid_parameter_value(
+                f"Invalid filter string: {filter_string}. Expected a single SQL expression.",
+            )
+
+        class Entity(Enum):
+            ATTRIBUTE = "attributes"
+            METRIC = "metrics"
+            PARAM = "params"
+            TAG = "tags"
+
+            def from_str(s: str) -> "Entity":
```

Comment:
Do we need this? We can use Entity(s) directly? (we can keep it for checking if s is allowed though)

Reply:
No, `Entity(s)` throws an error.